### PR TITLE
devicetree: add DT_CHILD()

### PR DESCRIPTION
**Pull request by galak**
Helper macro to get a node_id for a child node of a given node_id.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: include/devicetree.h [hunk 1]
```diff
@@ -208,6 +208,33 @@
...
+ *                     my-serial: serial@4 {
+ *                             status = "okay";
+ *                             current-speed = <115200>;
+ *                             ...
+ *                     };
+ *             };
+ *     };
+ *
+ * Example usage with @ref DT_PROP() to get the status of the child node
+ * "serial_4" of node referenced by node label "soc-label":
```
Comment (mbolivar-nordic) [inline include/devicetree.h:227]:
```suggestion
 * "serial@4" of the node referenced by node label "soc-label":
```

File: include/devicetree.h [hunk 2]
```diff
@@ -208,6 +208,33 @@
...
+ *
+ *     DT_PROP(DT_CHILD(DT_NODELABEL(soc_label), serial_4), status) // "okay"
+ *
+ * @param node_id node identifier
+ * @param child lowercase-and-underscores child node name
+ * @return node identifier for the node with the name refered to by 'child'
```
Comment (mbolivar-nordic) [inline include/devicetree.h:233]:
```suggestion
 * @return node identifier for the node with the name referred to by 'child'
```

File: include/devicetree.h [hunk 3]
```diff
@@ -208,6 +208,33 @@
...
+ */
+#define DT_CHILD(node_id, child) \
+	UTIL_CAT(node_id, MACRO_MAP_CAT(DT_S_PREFIX, child))
```
Comment (mbolivar-nordic) [inline include/devicetree.h:236]: Do you need the full MACRO_MAP_CAT? Its error messages are more confusing. Why not just `DT_S_PREFIX(child)`?

Comment (galak) [reply include/devicetree.h]: No reason, changed it to remove MACRO_MAP_CAT.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**